includes/api/class-wc-admin-rest-coupons-controller.php

```diff
@@ -24,4 +24,69 @@ class WC_Admin_REST_Coupons_Controller extends WC_REST_Coupons_Controller {
 	 */
 	protected $namespace = 'wc/v4';
 
+	/**
+	 * Get the query params for collections.
+	 *
+	 * @return array
+	 */
+	public function get_collection_params() {
+		$params                = parent::get_collection_params();
+		$params['search_code'] = array(
+			'description'       => __( 'Limit results to coupons with codes matching a given string.', 'woocommerce-admin' ),
+			'type'              => 'string',
+			'validate_callback' => 'rest_validate_request_arg',
+		);
+		return $params;
+	}
+
+
+	/**
+	 * Add coupon code searching to the WC API.
+	 *
+	 * @param WP_REST_Request $request Request data.
+	 * @return array
+	 */
+	protected function prepare_objects_query( $request ) {
+		$args = parent::prepare_objects_query( $request );
+
+		if ( ! empty( $request['search_code'] ) ) {
+			$args['search_code'] = $request['search_code'];
+		}
+
+		return $args;
+	}
+
+	/**
+	 * Get a collection of posts and add the code search option to WP_Query.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|WP_REST_Response
+	 */
+	public function get_items( $request ) {
+		add_filter( 'posts_where', array( __CLASS__, 'add_wp_query_search_code_filter' ), 10, 2 );
+		$response = parent::get_items( $request );
+		remove_filter( 'posts_where', array( __CLASS__, 'add_wp_query_search_code_filter' ), 10 );
+		return $response;
+	}
+
+	/**
+	 * Add code searching to the WP Query
+	 *
+	 * @param string $where Where clause used to search posts.
+	 * @param object $wp_query WP_Query object.
+	 * @return string
+	 */
+	public static function add_wp_query_search_code_filter( $where, $wp_query ) {
+		global $wpdb;
+
+		$code_search = $wp_query->get( 'search_code' );
+		if ( $code_search ) {
+			$code_search = $wpdb->esc_like( $code_search );
+			$code_search = ' \'%' . $code_search . '%\'';
```

Comment:
Nitpick: using `"` here would avoid escape characters.

Reply:
Added double quotes here as well as other PRs that override search params.